include/multipass/platform_win.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_PLATFORM_WIN_H
+#define MULTIPASS_PLATFORM_WIN_H
+
+#include <windows.h>
+
+#include <fmt/format.h>
+
+#include <string>
+
+struct WSAData;
+
+namespace multipass::platform
+{
+struct wsa_init_wrapper
+{
+    wsa_init_wrapper();
+    ~wsa_init_wrapper();
+
+    /**
+     * Check whether WSA initialization has succeeded.
+     *
+     * @return true WSA is initialized successfully
+     * @return false WSA initialization failed
+     */
+    operator bool() const noexcept
+    {
+        return wsa_init_result == 0;
+    }
+
+private:
+    WSAData* wsa_data{nullptr};
+    const int wsa_init_result{-1};
+};
+
+} // namespace multipass::platform
+
+/**
+ * Formatter for GUID type
+ */
+template <typename Char>
+struct fmt::formatter<::GUID, Char>
+{
+    constexpr auto parse(basic_format_parse_context<Char>& ctx)
+    {
+        return ctx.begin();
+    }
+
+    template <typename FormatContext>
+    auto format(const ::GUID& guid, FormatContext& ctx) const
+    {
+        // The format string is laid out char by char to allow it
+        // to be used for initializing variables with different character
+        // sizes.
+        static constexpr Char guid_f[] = {
```

Comment:
Maybe we can use MULTIPASS_UNIVERSAL_STRING here instead.

Reply:
I have no clue, but it's a huge oversight. I have found the MULTIPASS_UNIVERSAL_STRING workaround, which essentially defines two strings (one string, one wstring) for the given string using some macro trickery. Other than that, there's always the possibility of re-encoding the string in the target charset, but it wouldn't be trivial.